### PR TITLE
`app_service` - API update to `2022-09-01` 

### PR DESCRIPTION
**Pull request by jackofallops**
This PR collects together several fixes from other PRs that had significant conflicts introduced by a refactor in how `site_config` is processed. 

Supersedes #20350
Supersedes #21408

* Dependencies: update App Service to API version `2022-09-01`

* `azurerm_linux_function_app` - `allowed_origins` in the `cors` block now has a minimum entry count of `1` 
* `azurerm_linux_function_app` - added support for `public_network_access_enabled` property
* `azurerm_linux_function_app_slot` - `allowed_origins` in the `cors` block now has a minimum entry count of `1` 
* `azurerm_linux_function_app_slot` - added support for `public_network_access_enabled` property
* `azurerm_linux_web_app` - `allowed_origins` in the `cors` block now has a minimum entry count of `1` 
* `azurerm_linux_web_app` - added support for `public_network_access_enabled` property
* `azurerm_linux_web_app_slot`  - `allowed_origins` in the `cors` block now has a minimum entry count of `1` 
* `azurerm_linux_web_app_slot`  - added support for `public_network_access_enabled` property
* `azurerm_windows_function_app` - `allowed_origins` in the `cors` block now has a minimum entry count of `1` 
* `azurerm_windows_function_app` - added support for `public_network_access_enabled` property
* `azurerm_windows_function_app_slot` - `allowed_origins` in the `cors` block now has a minimum entry count of `1` 
* `azurerm_windows_function_app_slot` - added support for `public_network_access_enabled` property
* `azurerm_windows_web_app` - `allowed_origins` in the `cors` block now has a minimum entry count of `1` 
* `azurerm_windows_web_app` - added support for `public_network_access_enabled` property
* `azurerm_windows_web_app_slot` - `allowed_origins` in the `cors` block now has a minimum entry count of `1` 
* `azurerm_windows_web_app_slot` - added support for `public_network_access_enabled` property

### CODE REVIEW

File: internal/services/appservice/helpers/enums.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package helpers
+
+const (
+	PublicNetworkAccessEnabled  string = "Enabled"
+	PublicNetworkAccessDisabled string = "Disabled"
+)
```
Comment (tombuildsstuff) [inline internal/services/appservice/helpers/enums.go:6]: Is there a Swagger issue tracking these being omitted?

File: internal/services/appservice/linux_function_app_resource.go [hunk 2]
```diff
@@ -491,9 +498,19 @@ func (r LinuxFunctionAppResource) Create() sdk.ResourceFunc {
...
+					VnetRouteAllEnabled:  siteConfig.VnetRouteAllEnabled,
 				},
 			}
 
+			pna := helpers.PublicNetworkAccessEnabled
+			if !functionApp.PublicNetworkAccess {
+				pna = helpers.PublicNetworkAccessDisabled
+			}
+
+			// (@jackofallops) - Values appear to need to be set in both SiteProperties and SiteConfig for now?
```
Comment (tombuildsstuff) [inline internal/services/appservice/linux_function_app_resource.go:510]: is there a Swagger issue for this?

**Review by tombuildsstuff — APPROVED** [review]
A couple of comments but otherwise 👍

File: internal/services/appservice/helpers/enums.go (hunk 1 above)
Comment (jackofallops) [reply internal/services/appservice/helpers/enums.go:9]: Comment added to the tracking issue 👍

File: internal/services/appservice/linux_function_app_resource.go (hunk 2 above)
Comment (jackofallops) [reply internal/services/appservice/linux_function_app_resource.go]: Added 👍

**Comment by xanderdeseyn** [comment]
Any idea when we can expect 3.64.0?

**Comment by MattJeanes** [comment]
@xanderdeseyn the v3.64.0 milestone says due by July 06, 2023 and it's 100% complete so hopefully very soon!

image

**Comment by MattJeanes** [comment]
@xanderdeseyn I have noticed v3.64.0 is now live if you haven't seen: https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v3.64.0

**Comment by njglenuk** [comment]
It seems that the provider details for version 3.64 for linux_function_app on the Terraform.io page is missing the newly released enhancement for `public_network_access_enabled`:
https://registry.terraform.io/providers/hashicorp/azurerm/3.64.0/docs/resources/linux_function_app

I suspect the default for this argument is set to `true`?

**Comment by amrut057** [comment]
public_network_access_enabled doesn't seem to work in 3.64.0
Geting error: An argument named "public_network_access_enabled" is not expected here.

**Comment by eehret** [comment]
Can confirm that documentation is not up to date for some of these changes.

**Pull request merged**